### PR TITLE
fix(recorder): reattach toolbar if it was unmounted by framework hydration

### LINKED ISSUE #32632: [Bug]: Recorder Toolbar Hydration Problem
### Version

1.47.1

### Steps to reproduce

1. Clone my repo (`git clone git@github.com:akoenig/repro-playwright-hydration-error.git`)
2. Run `npm install && npm run dev`
3. Record a test via the VS Code extension.
4. Open up http://localhost:5173 in the appearing Chrome instance.
5. See the toolbar disappear.

### Expected behavior

I expect to still see the "recorder toolbar"

CleanShot 2024-09-16 at 10 26 05@2x

### Actual behavior

The "recorder toolbar" disappears when opening the Remix application.

### Additional context

[Video](https://share.cleanshot.com/vdgbMDsF)

### Environment

```shell
System:
    OS: macOS 14.5
    CPU: (12) arm64 Apple M2 Pro
    Memory: 187.00 MB / 32.00 GB
  Binaries:
    Node: 22.3.0 - ~/.local/share/devbox/global/default/.devbox/nix/profile/default/bin/node
    npm: 10.8.1 - ~/.local/share/devbox/global/default/.devbox/nix/profile/default/bin/npm
    pnpm: 9.4.0 - ~/.local/share/devbox/global/default/.devbox/nix/profile/default/bin/pnpm
    bun: 1.1.20 - ~/.local/share/devbox/global/default/.devbox/nix/profile/default/bin/bun
  IDEs:
    VSCode: 1.93.1 - /usr/local/bin/code
  Languages:
    Bash: 3.2.57 - /bin/bash
  npmPackages:
    @playwright/test: ^1.47.1 => 1.47.1
```

### PR DESCRIPTION
**Pull request by Skn0tt**
Closes https://github.com/microsoft/playwright/issues/32632. A side effect of Remix's hydration implementation is that it throws away the entire DOM. This is broadly discussed in https://github.com/remix-run/remix/issues/4822 - there might be a fix in coming React versions, but who knows.
Besides breaking browser extensions, this also deletes our toolbar!
This PR fixes it by periodically checking in on `x-pw-glass`, and remounting it if it was unmounted. Hacky but effective!

### CODE REVIEW

File: packages/playwright-core/src/server/injected/highlight.ts [hunk 1]
```diff
@@ -91,6 +92,11 @@ export class Highlight {
 
   install() {
     this._injectedScript.document.documentElement.appendChild(this._glassPaneElement);
+
+    this._glassPaneInterval = setInterval(() => {
```
Comment (Skn0tt) [inline packages/playwright-core/src/server/injected/highlight.ts:96]: I've tried using `MutationObserver` as the more canonical solution over `setInterval`, but couldn't get that to work.

**Review by dgozman — COMMENTED** [review]
This seems to be only relevant to the recorder. Should we put the code there instead?

**Review by dgozman — APPROVED** [review]

**Comment by akoenig** [comment]
@Skn0tt, thanks a lot for this change. How can I give it a test drive?

**Comment by Skn0tt** [comment]
We publish a nightly release under the `next` label: https://www.npmjs.com/package/playwright/v/next
So try `npm i playwright@next` tomorrow to test drive.

**Comment by akoenig** [comment]
@Skn0tt, pretty sure I miss something, but I still receive:

CleanShot 2024-09-18 at 08 16 17@2x

I had `@playwright/test` in my dependencies. That is why i updated this package to `1.48.0-alpha-2024-09-18`.

**Comment by Skn0tt** [comment]
Right, you won't be able to get around that. It's just a consequence of how React's rehydration works. Does this cause any errors except for this warning though?

**Comment by akoenig** [comment]
@Skn0tt, the toolbar still disappears in the current alpha release. Unfortunately, the recent bugfix didn't resolve this issue (at least for me in my reproduction repository).

**Pull request merged**